flink-end-to-end-tests/flink-end-to-end-tests-common/pom.xml

```diff
@@ -55,6 +55,12 @@ under the License.
 			<artifactId>flink-test-utils-junit</artifactId>
 			<scope>compile</scope>
 		</dependency>
+		<dependency>
+			<groupId>org.apache.flink</groupId>
+			<artifactId>flink-runtime_${scala.binary.version}</artifactId>
+			<version>${project.version}</version>
+			<scope>compile</scope>
+		</dependency>
```

Comment:
Which change made it necessary to add `flink-runtime` as a compile time dependency? Asked differently, where does `PrometheusReporterEndToEndITCase` need the `org.apache.flink.runtime.rest.messages.RequestBody`?

Reply:
It doesn't, at least not directly.

`flink-end-to-end-tests-common` requires flink-runtime for some REST stuff, but has no direct dependency; it only has the `provided` flink-dist and uses the transitive `flink-runtime` dependency, which is then also set to `provided`.

The shade-plugin then creates a fairly wacky dependency reduced pom; transitive `provided` dependencies result in an exclusion for said dependency, making it appear as if `flink-end-to-end-tests-common` does not have a dependency o `flink-runtime` anymore.

Excerpt from the dependeny-reduced-pom.xml:
```
<dependency>
      <groupId>org.apache.flink</groupId>
      <artifactId>flink-dist_2.11</artifactId>
      <version>1.11-SNAPSHOT</version>
      <scope>provided</scope>
      <exclusions>
        ...
        <exclusion>
          <artifactId>flink-runtime_2.11</artifactId>
          <groupId>org.apache.flink</groupId>
        </exclusion>
        ...
```

Then, when the e2e test runs and tries to use REST stuff the classes obviously aren't on the classpath.

The new runtime dependency ensures that this dependency is retained after shading.

Reply:
As for why this was discovered now, when the prometheus/metrics e2e tests were working against the `FlinkDistribution` they weren't reliant on any REST calls, and didn't run into the issue. After the migration to the `FlinkResource` they implicitly gained a dependency on REST calls, hence the problem.
Existing users of the `FlinkResource`, (i.e., kafka) have a separate runtime dependencies pulled in transitively via some other dependency.